Develop tl721x retention

haiwentelink opened the pull request.

s07641069 commented on CMakeLists.txt:
```diff
@@ -17,7 +17,11 @@ zephyr_link_libraries(${CMAKE_CURRENT_SOURCE_DIR}/zephyr/blobs/lib_zephyr_b95.a)
 elseif (CONFIG_SOC_RISCV_TELINK_TL321X)
 zephyr_link_libraries(${CMAKE_CURRENT_SOURCE_DIR}/zephyr/blobs/lib_zephyr_tl321x.a)
 elseif (CONFIG_SOC_RISCV_TELINK_TL721X)
+	if(CONFIG_PM)
+zephyr_link_libraries(${CMAKE_CURRENT_SOURCE_DIR}/zephyr/blobs/lib_zephyr_tl721x_pm.a)
+	else()
 zephyr_link_libraries(${CMAKE_CURRENT_SOURCE_DIR}/zephyr/blobs/lib_zephyr_tl721x.a)
+	endif()
```
After merging of http://192.168.48.36/src/open_platform/telink_b9x_ble_src/merge_requests/33
I think we can use lib_zephyr_tl721x.a only and revert this file changes

haiwentelink replied:
@s07641069  the background is , for pm and no pm which will have different library , we will add different efuse for different packaged soc in the future . follow as the sales purpose.

s07641069 commented on tlsr9/CMakeLists.txt:
```diff
@@ -93,7 +93,7 @@ if(CONFIG_GPIO_TELINK_B9X OR CONFIG_GPIO_TELINK_TLX)
 
 # PWM driver reference sources
-if(CONFIG_PWM_TELINK_B9X OR CONFIG_PWM_TELINK_TLX)
+if(CONFIG_PWM_TELINK_B9X OR CONFIG_PWM_TELINK_TLX OR CONFIG_SOC_RISCV_TELINK_TL721X)
```
why pwm.c is needed by default for CONFIG_SOC_RISCV_TELINK_TL721X ?
Maybe better to resolve compilation issue in case if such exist without adding pwm.c by default

haiwentelink replied:
the pm wakeup need pwm signal to trigger , so pwm is forced for tercel .

s07641069 commented on zephyr/module.yml:
```diff
@@ -36,9 +36,17 @@ blobs:
-    sha256: 645d1f948a70fa70f0f977c32390ad98ff458c8c0d29ca9eb7f1311b90867799
+    sha256: a6a2e54fa93b5bf6808e3b79b05229d7a16e11e703cbd151fe6f85b74aa03b06
     type: lib
     version: 'V4.0.4.2-c8c196e0'
     license-path: tlsr9/ble/license.txt
-    url: http://wiki.telink-semi.cn/wiki/protocols/Zephyr/binaries/public/tl721x_lib/lib_zephyr_tl721x_404e51e546195e114e4780b69956fe43f635c4f5.a
-    description: "Binary libraries supporting Telink TL721X series BLE subsystems"
\ No newline at end of file
+    url: http://wiki.telink-semi.cn/wiki/protocols/Zephyr/binaries/public/tl721x_lib/lib_zephyr_tl721x_694382dbee377cc85ad0edef52d7538debd0e595.a
+    description: "Binary libraries supporting Telink TL721X series BLE subsystems"
+
+  - path: lib_zephyr_tl721x_pm.a
+    sha256: 36511a53653e0379c4da1470b616e882bf5c129a0bbcee8496349e7218a65a38
+    type: lib
+    version: 'V4.0.4.2-c8c196e0'
+    license-path: tlsr9/ble/license.txt
+    url: http://wiki.telink-semi.cn/wiki/protocols/Zephyr/binaries/public/tl721x_lib/lib_zephyr_tl721x_694382dbee377cc85ad0edef52d7538debd0e595_pm.a
+    description: "Binary libraries supporting Telink TL721X series BLE subsystems, optimized for low-power mode"
```
Same as above after merging of BLE PR we may use single library

haiwentelink replied:
@s07641069 the background is , for pm and no pm which will have different library , we will add different efuse for different packaged soc in the future . follow as the sales purpose.

haiwentelink committed:
hal: telink: tl721x: update HAL for PM

- add PWM module into build for each app.
- add blc_ll_initPowerManagement_module for PM.
- add API definition to avoid build warnings and errors.
- update BLE lib for PM.
- add exit latency mechanism.

Signed-off-by: Haiwen Xia <haiwen.xia@telink-semi.com>

haiwentelink commented:
@s07641069  already merged zhihan's PR , now commission is more stable .
